x-conduit/src/X/Control/Monad/Trans/Resource.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module X.Control.Monad.Trans.Resource where
+
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Resource (MonadResource, liftResourceT)
+import Control.Monad.Trans.Either (EitherT)
+
+-- | This instance is useful when adding errors to the monad used for some conduit processing
+-- when resources, like files, are involved. For example:
+--
+-- foo :: MonadResource m => ConduitM () Void (EitherT Text m) [ByteString]
+-- foo = CB.sourceFile "src/Main.hs"
+--  =$= CB.lines
+--  -- filter some lines
+--  =$= CL.map parse
+--  -- collect errors in EitherT Text m
+--  =$= CL.mapM hoistEither
+--  =$= CL.consume
+--
+-- parse :: ByteString -> Either Text ByteString
+-- parse _ = Left ":("
+instance MonadResource m => MonadResource (EitherT e m) where
```

Comment:
Sorry, my suggestion, would be happy to do it a different way if it's possible, this is a full example using `MonadResource`, maybe you can show how to do it differently:

```haskell
{-# LANGUAGE OverloadedStrings #-}

import           Control.Monad.Trans (lift)
import           Control.Monad.Trans.Class
import           Control.Monad.Trans.Either
import           Control.Monad.Trans.Resource

import           Data.ByteString (ByteString)
import           Data.Conduit
import qualified Data.Conduit.Binary as CB
import qualified Data.Conduit.List as CL
import           Data.Text (Text)
import qualified Data.Text as T
import           Data.Void (Void)

main :: IO ()
main = do
  xs <- runResourceT (runEitherT (runConduit foo))
  print xs
  return ()

foo :: MonadResource m => ConduitM () Void (EitherT Text m) [ByteString]
foo = CB.sourceFile "src/Main.hs"
  =$= CB.lines
  =$= CL.map parse
  =$= CL.mapM hoistEither
  =$= CL.consume

parse :: ByteString -> Either Text ByteString
parse _ = Left ":("

instance MonadResource m => MonadResource (EitherT e m) where
  liftResourceT = lift . liftResourceT
```